core/drivers/qcom_geni_uart.c

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Qualcomm GENI serial engine UART driver
+ *
+ * Copyright (c) 2025, Linaro Limited
+ * Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+ */
+
+#include <drivers/qcom_geni_uart.h>
+#include <io.h>
+
+#define GENI_STATUS_REG			0x40
+#define GENI_STATUS_REG_CMD_ACTIVE	BIT(0)
+#define GENI_TX_FIFO_REG		0x700
+#define GENI_TX_TRANS_LEN_REG		0x270
+#define GENI_M_CMD0_REG			0x600
+
+#define GENI_M_CMD_TX			0x8000000
+
+static void qcom_geni_uart_putc(struct serial_chip *chip, int ch)
+{
+	struct qcom_geni_uart_data *pd =
+		container_of(chip, struct qcom_geni_uart_data, chip);
+	vaddr_t base = io_pa_or_va(&pd->base, GENI_UART_REG_SIZE);
+	uint64_t timer = timeout_init_us(1000 * 1000);
+
+	while (!timeout_elapsed(timer) &&
```

Comment:
can we not drop the time out to keep the function simple? seems to be the consensus around these type of drivers